### PR TITLE
Change server folder structure and refactor code

### PR DESCRIPTION
**Pull request by carlodmendoza**

### CODE REVIEW

File: final-project/README.md [hunk 1]
```diff
@@ -5,6 +5,45 @@ This application allows users to sign up for an account, sign in, record and man
...
+    │   │   
+    │   └───filebased/          - package for file-based DB implementation
+    │   │   
+    │   └───redis/              - package for Redis store implementation  
+    │   │
+    │   └───data.go             - file for data layer interface implementation 
+    │
+    │   sessions/               - package for handling user sessions
+    │   transactions/           - package for handling user transactions
+    └───Dockerfile
```
Comment (florentsolt) [inline final-project/README.md]: I suggest to move that Dockerfile in "/deploy".
At the moment we don't know it will be the same file dev/stage/prod.

File: final-project/README.md [hunk 2]
```diff
@@ -5,6 +5,45 @@ This application allows users to sign up for an account, sign in, record and man
...
+│       │   └───data.json       - sample data for testing dev
+│       │ 
+│       │   .env                - environment variables for dev
+│       └───docker-compose.yml
+│   
+└───server/
+    │   auth/                   - package for user registration and authentication
+    │   categories/             - package for handling categories of transactions
+    │
+    └───data/
```
Comment (florentsolt) [inline final-project/README.md]: Why not "storage"
I always try to avoid "data", it means everything and it's never explicit.

File: final-project/README.md [hunk 3]
```diff
@@ -5,6 +5,45 @@ This application allows users to sign up for an account, sign in, record and man
...
+│   README.md
+│   routes.go
+│
+└───client/                     - module for client app
+│   
+└───deploy/
+│   │
+│   └───dev/
+│       │   
+│       └───data/
```
Comment (florentsolt) [inline final-project/README.md]: You may have to have some files for redis and some files for the "file" storage.
But it's not important.

File: final-project/README.md [hunk 4]
```diff
@@ -5,6 +5,45 @@ This application allows users to sign up for an account, sign in, record and man
...
 The server and client are both written on Go, and run on Docker.
 
+## Project Directory
+```
+final-project/
+│   Taskfile.yml
+│   go.mod
+│   main.go                     - main file for running the server
+│   README.md
+│   routes.go
```
Comment (florentsolt) [inline final-project/README.md]: Move main.go and routes.go in the "server" folder.
Those files are not really related to the client.

File: final-project/README.md [hunk 5]
```diff
@@ -5,6 +5,45 @@ This application allows users to sign up for an account, sign in, record and man
 
 The server and client are both written on Go, and run on Docker.
 
+## Project Directory
+```
+final-project/
+│   Taskfile.yml
```
Comment (florentsolt) [inline final-project/README.md]: It's not because I've suggest and also using it, that it's a good fit for you.
Some people are using a Makefile, or some shell scripts, ...
If you want to use it, you will have to justify why it's a good choice for you :)

File: final-project/README.md (hunk 1 above)
Comment (carlodmendoza) [reply final-project/README.md]: My client app also has its own Dockerfile. Do I also move it to `deploy/`?
If yes, can the structure be like this:
In `deploy/dev/`, I have `client/` and `server/`. Dockerfile of client goes to `client/`, while Dockerfile of server and `data/` goes to `server/`

Comment (florentsolt) [reply final-project/README.md]: Yep you can do that, of course.
